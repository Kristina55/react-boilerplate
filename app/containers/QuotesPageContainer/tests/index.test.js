/**
 *
 * Tests for QuotesPageContainer
 *
 *
 */
import React from 'react';
import { browserHistory } from 'react-router-dom';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import QuoteListPage from '../../../components/QuotesListPage';
import configureStore from '../../../configureStore';

describe('QuoteListPage', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const quotes = ['mock1', 'mock2', 'mock3'];
    const loadingQuotes = jest.fn();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <QuoteListPage fetchQuotes={loadingQuotes} quotes={quotes} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  it('should load stings from server on mount', () => {
    const quotes = ['mock1', 'mock2', 'mock3'];
    const loadingQuotesSpy = jest.fn();
    render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <QuoteListPage quotes={quotes} fetchQuotes={loadingQuotesSpy} />
        </IntlProvider>
      </Provider>,
    );
    expect(loadingQuotesSpy).toHaveBeenCalled();
  });
});
