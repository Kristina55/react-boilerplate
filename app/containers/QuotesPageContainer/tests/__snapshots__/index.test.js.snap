// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteListPage should render and match the snapshot 1`] = `
<div
  class="QuotesListPage__QuoteWrapper-sc-1mzta8u-0 fUILbL"
>
  <div
    class="QuotesListPage__Quotes-sc-1mzta8u-1 isPNRt"
  >
    <ul>
      <div>
        <li
          class="Quote__QuoteStyle-sc-5smwmg-0 wKlZm"
        >
          mock1
        </li>
      </div>
      <div>
        <li
          class="Quote__QuoteStyle-sc-5smwmg-0 wKlZm"
        >
          mock2
        </li>
      </div>
      <div>
        <li
          class="Quote__QuoteStyle-sc-5smwmg-0 wKlZm"
        >
          mock3
        </li>
      </div>
    </ul>
  </div>
</div>
`;
